### DISCUSSION TITLE
How can I use internationalization without middleware?

### DISCUSSION THREAD

**Discussion by thedv91** [vercel/next.js#52529] [App Router]
I don't want to waste cost using middleware just to redirect and check the URL. Is there any way I can implement it like in the `pages` folder?

**Comment by TheCodby** [comment]
As far as I know, it is currently not possible to use `setHeader` in the app directory like it is done in the page directory. So you can't

**Reply by thedv91** [reply]
I think App Router does not support i18n config in `next.config.js` in the same way as `pages`

**Comment by cezarneaga** [comment]
hey everyone, 

it must be one of those things where an elegant solution was the simplest to implement and given a generous number of edge middleware invocations given made sense in a time constraint env.

but in the interest of adopting server components and upgrading larger apps to the app server, a solution for large websites should be available in my opinion. perhaps is just a matter of us requesting it and nextjs team having a bit more of a headspace. 

as such, I would like to have this option for our team.

marry holidays,
cezar
